src/compile/selection/parse.ts

```diff
@@ -52,9 +52,10 @@ export function parseUnitSelection(model: UnitModel, selDefs: SelectionParameter
       events: isString(defaults.on) ? parseSelector(defaults.on, 'scope') : array(duplicate(defaults.on))
     } as any);
 
+    const def_ = duplicate(def); // defensive copy to prevent compilers from causing side effects
```

Comment:
Yes, this PR introduces side effects — namely, if the interval selection parser detects a map projection, it rewrites the defaults from `x/y` projection to instead `lat/lon` and updates the predicate field to `SELECTION_ID` ready for subsequent selection parsers to operate. This was the cleanest, most scoped/localized way I could think of to introduce brushing on maps without triggering a more massive refactor since the rest of the interval selection components expect standard cartesian coordinate systems. So my preference would be to keep the defensive copy since its occurring just on the selection definitions.